inc/classes/CDN/RocketCDN/APIClient.php

```diff
@@ -278,4 +278,41 @@ public function update_website_status( $is_active ) {
 			$args
 		);
 	}
+
+	/**
+	 * Filter the arguments used in an HTTP request, to make sure our user token has not been overwritten
+	 * by some other plugin.
+	 *
+	 * @since  3.5
+	 * @author Rémy Peorna
+	 *
+	 * @param  array  $args An array of HTTP request arguments.
+	 * @param  string $url  The request URL.
+	 * @return array
+	 */
+	public function preserve_authorization_token( $args, $url ) {
+		if ( strpos( $url, self::ROCKETCDN_API ) === false ) {
+			return $args;
+		}
+
+		if ( empty( $args['headers']['Authorization'] ) ) {
+			return $args;
+		}
```

Comment:
So if a plugin completely removes the header...
Shouldn't we add it back anyway?